pkg/registry/apis/dashboard/v1alpha1/register.go

```diff
@@ -217,6 +223,40 @@ func (b *DashboardsAPIBuilder) PostProcessOpenAPI(oas *spec3.OpenAPI) (*spec3.Op
 	return oas, nil
 }
 
+// Validate will prevent deletion of provisioned dashboards, unless the grace period is set to 0, indicating a force deletion
+func (b *DashboardsAPIBuilder) Validate(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces) (err error) {
```

Comment:
The duplicate code is getting a bit overwehelming 😬  

Maybe we can create `DashboardsAPIBuilderBase` struct (or similar) that the common functions and then we only need to implement once?